pkg/config/validation/virtualservice.go

```diff
@@ -142,17 +142,31 @@ func validateHTTPRouteMatchRequest(http *networking.HTTPRoute, routeType HTTPRou
 					if header == nil {
 						errs = appendErrors(errs, fmt.Errorf("header match %v cannot be null", name))
 					}
-
 					if _, ok := header.GetMatchType().(*networking.StringMatch_Prefix); ok {
 						if header.GetPrefix() == "" {
 							errs = appendErrors(errs, fmt.Errorf("header prefix match %v may not be empty", name))
 						}
 					}
-
 					errs = appendErrors(errs, ValidateHTTPHeaderName(name))
 					errs = appendErrors(errs, validateStringMatchRegexp(header, "headers"))
 				}
 
+				for name, header := range match.WithoutHeaders {
+					if header == nil {
+						errs = appendErrors(errs, fmt.Errorf("withoutHeaders match %v cannot be null", name))
```

Comment:
This is valid if you look at the processing in route.go. Though i donot know how we can set it nil with kubectl apply

Reply:
thanks, updated.

Reply:
unit tests covered.
```
validation: TestValidateHTTPRoute/null_header_match expand_less | 0s
-- | --
{Failed      validation_test.go:2934: got valid=true but wanted valid=false: }
````

Reply:
@hzxuzhonghu delete the unit test or revert https://github.com/istio/istio/pull/49608/commits/3951f47462e31538ad8ffbfecd0af80dace8f4a5 ?

Reply:
why not fix it